docs/source/contracts.dot

```diff
@@ -0,0 +1,50 @@
+digraph "Beamer contracts" {
```

Comment:
Can you maybe also mark chain-independent contracts and chain dependent contracts with different colors?

In general it would be nice to have clearly explained somewhere (in one or two sentences) that there are chain dependent contracts to work with their specific messengers (or more abstract contract interfaces) and chain independent ones which are purely dependent to our protocol. Basically a short description why we need to distinguish.

Reply:
I totally agree and I planned to write a bit more, but my time was eaten away by... more pressing concerns. :sweat_smile: 
Now that the Optimism E2E test is working, I can write some more.

Reply:
Done. Chain-dependent contracts are now colored differently. Also added tables listing all the contracts and the chains they are deployed to. Finally, some wording has been improved and a few sentences added to make things more clear.